persistence/sql/persister_oauth2.go

```diff
@@ -429,7 +455,34 @@ func (p *Persister) CreateRefreshTokenSession(ctx context.Context, signature str
 func (p *Persister) GetRefreshTokenSession(ctx context.Context, signature string, session fosite.Session) (request fosite.Requester, err error) {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.GetRefreshTokenSession")
 	defer otelx.End(span, &err)
-	return p.findSessionBySignature(ctx, signature, session, sqlTableRefresh)
+
+	r := OAuth2RefreshTable{OAuth2RequestSQL: OAuth2RequestSQL{Table: sqlTableRefresh}}
+	err = p.QueryWithNetwork(ctx).Where("signature = ?", signature).First(&r)
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, errorsx.WithStack(fosite.ErrNotFound)
+	} else if err != nil {
+		return nil, sqlcon.HandleError(err)
+	}
+
+	fositeRequest, err := r.toRequest(ctx, session, p)
+	if err != nil {
+		return nil, err
+	}
+
+	if r.Active {
+		return fositeRequest, nil
+	}
```

Comment:
I guess that we should avoid unnecessary decryption of the data if possible?

```suggestion
	if r.Active {
		return r.toRequest(ctx, session, p)
	}
```

Reply:
I see now, we have to return the request even with error conditions...